src/Build/Collections/CopyOnReadEnumerable.cs

```diff
@@ -48,37 +56,21 @@ public CopyOnReadEnumerable(IEnumerable<T> backingEnumerable, object syncRoot)
         /// Returns an enumerator over the collection.
         /// </summary>
         /// <returns>The enumerator.</returns>
-        public IEnumerator<T> GetEnumerator()
+        public IEnumerator<TResult> GetEnumerator()
         {
-            List<T> list;
+            List<TResult> list;
             if (_backingEnumerable is ICollection backingCollection)
             {
-                list = new List<T>(backingCollection.Count);
+                list = new List<TResult>(backingCollection.Count);
             }
             else
             {
-                list = new List<T>();
+                list = new List<TResult>();
             }
 
-            bool isCloneable = false;
-            bool checkForCloneable = true;
             lock (_syncRoot)
             {
-                foreach (T item in _backingEnumerable)
-                {
-                    if (checkForCloneable)
-                    {
-                        if (item is IDeepCloneable<T>)
-                        {
-                            isCloneable = true;
-                        }
-
-                        checkForCloneable = false;
-                    }
-
-                    T copiedItem = isCloneable ? (item as IDeepCloneable<T>).DeepClone() : item;
-                    list.Add(copiedItem);
-                }
+                list.AddRange(_backingEnumerable.Select(_selector));
```

Comment:
Is this another case of using x => func(x) being faster than just func?

Reply:
Select takes a `Func<T, TResult>` so I imagine it would be fine in this case. The caller is responsible for lambda/delegate to func conversion.